### ISSUE TITLE
Following MVC Tutorial Produces Read Permissions Error

### ISSUE THREAD

**Issue by QuantumBits** [GenieFramework/Genie.jl#490]
**Describe the bug**
Following [this tutorial](https://genieframework.github.io/Genie.jl/dev/tutorials/4-1--Developing_MVC_Web_Apps.html#Getting-started-creating-the-app) produces this error: `SystemError: opening file "db/connection.yml": Permission denied`

**Error stacktrace**
```julia
julia> Genie.newapp_mvc("Watch tonight")

Please choose the DB backend you want to use: 
1. SQLite
2. MySQL
3. PostgreSQL
4. Skip installing DB support at this time

Input 1, 2, 3 or 4 and press ENTER to confirm.
If you are not sure what to pick, choose 1 (SQLite). It is the simplest option to get you started right away.
You can add support for additional databases anytime later. 

1
  Activating new project at `~/Code/Julia/Genie`
[ Info: Done! New app created at ~/Code/Julia/Genie/Watchtonight
[ Info: Changing active directory to ~/Code/Julia/Genie/Watchtonight
    /tmp/jl_QdeQCl/Watchtonight/Project.toml
[ Info: Project.toml has been generated
[ Info: Installing app dependencies
  Activating project at `~/Code/Julia/Genie/Watchtonight`
  Activating project at `~/Code/Julia/Genie/Watchtonight`
[ Info: Installing dependencies for unit tests
  Activating new project at `~/Code/Julia/Genie/Watchtonight/test`
  Activating project at `~/Code/Julia/Genie/Watchtonight`
[ Info: Starting your brand new Genie app - hang tight!
ERROR: SystemError: opening file "db/connection.yml": Permission denied
Stacktrace:
  [1] systemerror(p::String, errno::Int32; extrainfo::Nothing)
    @ Base ./error.jl:174
  [2] #systemerror#68
    @ ./error.jl:173 [inlined]
  [3] systemerror
    @ ./error.jl:173 [inlined]
  [4] open(fname::String; lock::Bool, read::Nothing, write::Nothing, create::Nothing, truncate::Bool, append::Nothing)
    @ Base ./iostream.jl:293
  [5] open(fname::String, mode::String; lock::Bool)
    @ Base ./iostream.jl:355
  [6] open(fname::String, mode::String)
    @ Base ./iostream.jl:355
  [7] open(::var"#1#2", ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:328
  [8] open(::Function, ::String, ::String)
    @ Base ./io.jl:328
  [9] top-level scope
    @ ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:573
 [10] eval
    @ ./boot.jl:373 [inlined]
 [11] loadapp(path::String; autostart::Bool, dbadapter::String)
    @ Genie ~/.julia/packages/Genie/ZkOSx/src/Genie.jl:198
 [12] autostart_app(path::String; autostart::Bool, initdb::Bool, dbadapter::String)
    @ Genie.Generator ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:592
 [13] post_create(app_name::String, app_path::String; autostart::Bool, testmode::Bool, dbsupport::Bool, dbadapter::String, interactive::Bool)
    @ Genie.Generator ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:704
 [14] newapp(app_name::String; autostart::Bool, fullstack::Bool, dbsupport::Bool, mvcsupport::Bool, testmode::Bool, dbadapter::Nothing, interactive::Bool)
    @ Genie.Generator ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:684
 [15] #newapp_mvc#45
    @ ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:767 [inlined]
 [16] newapp_mvc(path::String)
    @ Genie.Generator ~/.julia/packages/Genie/ZkOSx/src/Generator.jl:767
 [17] top-level scope
    @ REPL[2]:1

shell> ls -la db
total 20
drwxr-xr-x 4 user domain users 4096 Feb 16 10:44 .
drwxr-xr-x 9 user domain users 4096 Feb 16 10:44 ..
-r--r--r-- 1 user domain users  260 Feb 16 10:44 connection.yml
drwxr-xr-x 2 user domain users 4096 Feb 16 10:44 migrations
drwxr-xr-x 2 user domain users 4096 Feb 16 10:44 seeds
```

**To reproduce**
Installed Genie.jl v4.12.0 and SearchLight.jl v2.2.0 to the "base" environment of my Julia 1.7.2 installation. Then ran the following to produce the above.

```julia
julia> using Genie
julia> Genie.newapp_mvc("Watch tonight")
```

**Expected behavior**
The error goes away when I run julia with "sudo" / as root:

```bash
user@computer:~/Code/Julia/Genie$ sudo julia
[sudo] password for user:
```

```julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.2 (2022-02-06)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Genie

julia> Genie.newapp_mvc("Watch tonight")


Please choose the DB backend you want to use: 
1. SQLite
2. MySQL
3. PostgreSQL
4. Skip installing DB support at this time

Input 1, 2, 3 or 4 and press ENTER to confirm.
If you are not sure what to pick, choose 1 (SQLite). It is the simplest option to get you started right away.
You can add support for additional databases anytime later. 

1
  Activating project at `~/Code/Julia/Genie`
[ Info: Done! New app created at ~/Code/Julia/Genie/Watchtonight
[ Info: Changing active directory to ~/Code/Julia/Genie/Watchtonight
    /tmp/jl_IPvSFv/Watchtonight/Project.toml
[ Info: Project.toml has been generated
[ Info: Installing app dependencies
  Activating project at `~/Code/Julia/Genie/Watchtonight`
  Activating project at `~/Code/Julia/Genie/Watchtonight`
[ Info: Installing dependencies for unit tests
  Activating new project at `~/Code/Julia/Genie/Watchtonight/test`
  Activating project at `~/Code/Julia/Genie/Watchtonight`
[ Info: Starting your brand new Genie app - hang tight!

 _____         _
|   __|___ ___|_|___
|  |  | -_|   | | -_|
|_____|___|_|_|_|___|

| Web: https://genieframework.com
| GitHub: https://github.com/genieframework/Genie.jl
| Docs: https://www.genieframework.com/docs/tutorials/Overview.html
| Discord: https://discord.com/invite/9zyZbD6J7H
| Twitter: https://twitter.com/GenieMVC

Active env: DEV


Ready! 

┌ Info: 2022-02-16 10:40:13 
└ Web Server starting at http://127.0.0.1:8000 

julia> 

shell> ls -la db
total 20
drwxr-xr-x 4 root root 4096 Feb 16 10:40 .
drwxr-xr-x 9 root root 4096 Feb 16 10:39 ..
-r--r--r-- 1 root root   74 Feb 16 10:39 connection.yml
-rw-r--r-- 1 root root    0 Feb 16 10:40 dev.sqlite3
drwxr-xr-x 2 root root 4096 Feb 16 10:39 migrations
drwxr-xr-x 2 root root 4096 Feb 16 10:39 seeds
```

**Additional context**

```julia
julia> versioninfo()
Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake-avx512)
Environment:
  JULIA_PKG_USE_CLI_GIT = true
  JULIA_REVISE = auto

(Watchtonight) pkg> st
     Project Watchtonight v0.1.0
      Status `~/Code/Julia/Genie/Watchtonight/Project.toml`
  [c43c736e] Genie v4.12.0
  [6d011eab] Inflector v1.0.1
  [e6f89c97] LoggingExtras v0.4.7
  [739be429] MbedTLS v1.0.3
  [340e8cb6] SearchLight v2.2.0
  [21a827c4] SearchLightSQLite v2.1.0
  [ade2ca70] Dates
  [56ddb016] Logging

(Watchtonight) pkg> activate
  Activating project at `~/.julia/environments/v1.7`

(@v1.7) pkg> st
      Status `~/.julia/environments/v1.7/Project.toml`
  [c43c736e] Genie v4.12.0
  [340e8cb6] SearchLight v2.2.0
```

**Comment by QuantumBits** [comment]
Oh one last bit of information if it helps is I'm running on an AWS Linux workspace, essentially Centos/RHEL/Fedora.

**Comment by AbhimanyuAryan** [comment]
@QuantumBits I have tested Genie with Julia 1.7 on Debian ARM, m1, mac intel, windows. No issues on my side. Can you join our discord group to see if any other user with same OS is having this issue?

Also please follow the official new docs here: https://www.genieframework.com/docs/

**Comment by essenciary** [comment]
@AbhimanyuAryan I replicated the issue yesterday on macOS. It works now because I fixed it 12h ago :) It's now fixed in the latest v4.13.
